### ISSUE TITLE
binlog test is hard to interpret

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#719]
The binlog test creates a binary log, and then compares the result to a reference binary. This is an all-or-nothing test, which is hard to debug. If some bytes are different, it is hard to understand why and go to the root cause.

**Comment by steschu77** [comment]
I have a binlog to text conversion on my todo list. Would that be something you think would relax that problem?

**Comment by huitema** [comment]
As part of the last PR, I restore the return condition of the binlog test. It was always returning 0. Turns out that the binary comparison was failing. I had no idea why -- the binary file was shorter than expected. It may be because the input changed, or because I introduced a regression. Doing the comparison on some kind of text transform would be easier to diagnose. Can we just use one of the existing transforms, maybe qlog?

**Comment by steschu77** [comment]
Of course, we can use qlog as well. I'll prepare a PR for that.

**Comment by huitema** [comment]
In fact, bug #747 points to a design issue. The "log writer" contains its own binary frame parser, picoquic_binlog_frames. Each of the frame type contains form of the form "skip the frame to compute its length, then a call to `picoquic_binlog_frame(f, bytes_begin, bytes);`

There are two problems with having a specific parser: it could be wrong, and also it needs to be updated whenever the format changes. It would seem more appropriate to replace all this code by a simple loop on picoquic_skip_frame(), then a call to `binlog_frame`

**Comment by huitema** [comment]
On the other hand, when I tried to do what I proposed above, I came to appreciate that the log file only contains a summary of each frame, and that the problem is more complex than logging all the bytes in the frame. On the other hand, by doing the exercise and comparing the skip_log and basic_skip routines, I found a bug in the log parsing of the application frames. That bug would also have been found if we could review the results of the binlog test in a readable format.
